### DISCUSSION TITLE
Upgrading from 3.4.4 to 3.7.8: Error upgrade command not found

### DISCUSSION THREAD

**Discussion by lp-hdd** [netbox-community/netbox#16540] [Help Wanted!]
I am trying to get to version 4 eventually. I am not well versed in the Linux environment, just trying to upgrade our instance of NetBox.

Following the instructions on the upgrade page, everything checks out until I run the upgrade command:

**sudo PYTHON=/usr/bin/python3.12 ./upgrade.sh**

```
sudo: ./upgrade.sh: command not found 
```

The default version of python is 3.8, but I have installed 3.12 to run the upgrade. 

I found this command to try and I get this:

**PYTHON=/usr/bin/python3.12 /opt/netbox/upgrade.sh**

```
You are installing (or upgrading to) NetBox version 3.7.8
Using Python 3.12.4
Removing old virtual environment...
Creating a new virtual environment at /opt/netbox-3.7.8/venv...
Error: Command '['/opt/netbox-3.7.8/venv/bin/python3.12', '-m', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
--------------------------------------------------------------------
ERROR: Failed to create the virtual environment. Check that you have
the required system packages installed and the following path is
writable: /opt/netbox-3.7.8/venv
```

**Comment by zhazell** [comment]
I'm getting the same exact error trying to upgrade from 3.4.4 to 3.7.8. 

```You are installing (or upgrading to) NetBox version 3.7.8
Using Python 3.12.4
Removing old virtual environment...
Creating a new virtual environment at /opt/netbox-3.7.8/venv...
Error: Command '['/opt/netbox-3.7.8/venv/bin/python3', '-m', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
--------------------------------------------------------------------
ERROR: Failed to create the virtual environment. Check that you have
the required system packages installed and the following path is
writable: /opt/netbox-3.7.8/venv
--------------------------------------------------------------------```

**Reply by lp-hdd** [reply]
I got it working running the upgrade command in the directory: /opt/netbox-3.7.8/

**Reply by zhazell** [reply]
I was missing the extra python packages "python3-pip python3-venv python3-dev" so the upgrade script kept failing. I had to upgrade python from 3.9 to 3.12 and make sure I have the those extra packages for 3.12 and then the upgrade script worked.

**Comment by Supreme-Bob** [comment]
sudo apt-get install python3.12-venv python3.12-dev <- same issue had to install the venv & dev packages I had only installed the python3.12 package
